### PR TITLE
Reader style tweaks: register in Dispatcher manually

### PR DESCRIPTION
**Pull request by hius07**
Style tweaks can be applied with a gesture or added to a profile.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9816)

### CODE REVIEW

**Comment by poire-z** [comment]
You will solve https://www.mobileread.com/forums/showthread.php?p=4274863#post4274863 :) (to which I would have replied "yes, wrong tool for the job", but ok, no other tool available, so the one you just made will have to do).

File: frontend/apps/reader/modules/readerstyletweak.lua [hunk 1]
```diff
@@ -123,25 +124,36 @@ function TweakInfoWidget:init()
...
     local buttons = {
         {
-            text = _("Close"),
-            callback = function()
-                UIManager:close(self)
-            end,
+            {
+                text = self.is_tweak_registered and _("Don't show in action list") or _("Show in action list"),
+                callback = function()
+                    self.toggle_tweak_registered_callback()
```
Comment (poire-z) [inline frontend/apps/reader/modules/readerstyletweak.lua:130]: The wording "tweak registered" in this pure styletweaks context is a bit confusing - I would mention "dispatcher" in the variables/settings/methods, ie. `is_dispatcher_registered`.

File: frontend/apps/reader/modules/readerstyletweak.lua [hunk 2]
```diff
@@ -427,12 +439,23 @@ function ReaderStyleTweak:onSaveSettings()
...
     self.ui.doc_settings:saveSetting("book_style_tweak_enabled", self.book_style_tweak_enabled)
     self.ui.doc_settings:saveSetting("book_style_tweak_last_edit_pos", self.book_style_tweak_last_edit_pos)
 end
 
+local function dispatcherRegisterStyleTweak(tweak_id, tweak_title)
+    Dispatcher:registerAction("style_tweak_"..tweak_id,
+        {category="none", event="ToggleStyleTweak", arg=tweak_id, title=T(_("Toggle style tweak: %1"), tweak_title), rolling=true})
+end
+
+local function dispatcherRemoveStyleTweak(tweak_id)
```
Comment (poire-z) [inline frontend/apps/reader/modules/readerstyletweak.lua:453]: Wording only again: do we use Register/Remove elsewhere ? I would expect Register/Unregister.

File: frontend/apps/reader/modules/readerstyletweak.lua [hunk 3]
```diff
@@ -427,12 +439,23 @@ function ReaderStyleTweak:onSaveSettings()
...
     G_reader_settings:saveSetting("style_tweaks", self.global_tweaks)
+    G_reader_settings:saveSetting("style_tweaks_registered", self.registered_tweaks)
     self.ui.doc_settings:saveSetting("book_style_tweak", self.book_style_tweak)
     self.ui.doc_settings:saveSetting("book_style_tweak_enabled", self.book_style_tweak_enabled)
     self.ui.doc_settings:saveSetting("book_style_tweak_last_edit_pos", self.book_style_tweak_last_edit_pos)
 end
 
+local function dispatcherRegisterStyleTweak(tweak_id, tweak_title)
+    Dispatcher:registerAction("style_tweak_"..tweak_id,
+        {category="none", event="ToggleStyleTweak", arg=tweak_id, title=T(_("Toggle style tweak: %1"), tweak_title), rolling=true})
```
Comment (poire-z) [inline frontend/apps/reader/modules/readerstyletweak.lua:450]: Dunno if that's possible (or if all user-added settings and styletweaks are just non-contextualized stuff for Dispatcher), but seeing your screenshot, a separator between previous and all styletweaks would be nice.

**Comment by hius07** [comment]
Also some discussion in https://github.com/koreader/koreader/issues/8590#issuecomment-1260671499.

File: frontend/apps/reader/modules/readerstyletweak.lua (hunk 2 above)
Comment (hius07) [reply frontend/apps/reader/modules/readerstyletweak.lua:453]: https://github.com/koreader/koreader/blob/486d7071c7f08dca6b2d247c1d67dae1a615b123/plugins/profiles.koplugin/main.lua#L57-L64

Comment (poire-z) [reply frontend/apps/reader/modules/readerstyletweak.lua:453]: Didn't notice that at the time (or I was less interested by the profile plugin than by this styletweak module :).
I'd prefer to read all (these 2 only then?) as register/unregister - but up to you.

File: frontend/apps/reader/modules/readerstyletweak.lua (hunk 3 above)
Comment (hius07) [reply frontend/apps/reader/modules/readerstyletweak.lua:450]: Done

File: frontend/apps/reader/modules/readerstyletweak.lua (hunk 1 above)
Comment (hius07) [reply frontend/apps/reader/modules/readerstyletweak.lua:130]: It corresponds with `self.registered_tweaks` and `style_tweaks_registered` G_setting.

File: frontend/apps/reader/modules/readerstyletweak.lua (hunk 2 above)
Comment (hius07) [reply frontend/apps/reader/modules/readerstyletweak.lua]: Done

File: frontend/apps/reader/modules/readerstyletweak.lua (hunk 1 above)
Comment (poire-z) [reply frontend/apps/reader/modules/readerstyletweak.lua:130]: Yes, I saw that. But even `style_tweaks_registered` is a bit anonymous when just viewing the settings. If `style_tweaks_dispatcher_registered` is too long, just seeing "dispatcher" is better than "registered" (into what?). `style_tweaks_dispatcher` ? `style_tweaks_in_dispatcher` ?  so-so :/
`registered` sounds too alike `enabled` when just studying this readerstyletweak.lua module.

Comment (poire-z) [reply frontend/apps/reader/modules/readerstyletweak.lua:130]: Feels like `style_tweaks_in_dispatcher`, `is_tweak_in_dispatcher`, `self.tweaks_in_dispatcher` would read well.
If you really feel like sticking with "..._registered", please add some comment where this thing is created/read/saved, `-- registered in dispatcher`, so we don't have to follow it all the way to where it is neighboured by `dispatcherRegisterStyleTweak()` to know it is actually in Dispatcher that it is registered :)

Comment (hius07) [reply frontend/apps/reader/modules/readerstyletweak.lua:130]: I'm not stuck, just have no good idea. Let it be "in_dispatcher".

**Review by poire-z — APPROVED** [review]

**Pull request merged**